spyglass/server.py

```diff
@@ -74,6 +75,52 @@ def send_default_headers(self):
         def send_jpeg_content_headers(self, frame):
             self.send_header('Content-Type', 'image/jpeg')
             self.send_header('Content-Length', len(frame))
+            
+        def check_paths_match(self, expected_url, incoming_url):
+            
+            # Assign paths from URL into list
+            exp_paths = urlparse(expected_url.strip("/")).path.split("/")
+            inc_paths = urlparse(incoming_url.strip("/")).path.split("/")
+            
+            # Drop ip/hostname if present in path
+            if '.' in exp_paths[0]: exp_paths.pop(0)
+            if '.' in inc_paths[0]: inc_paths.pop(0)
+            
+            # Determine if match
+            if len(exp_paths)==len(inc_paths):
+                return all([exp == inc for exp, inc in zip(exp_paths, inc_paths)])
+            
+            return False
+        
+        def check_params_match(self, expected_url, incoming_url):
+            
+            # Check URL params
+            exp_params = parse_qsl(urlparse(expected_url).query)
+            inc_params = parse_qsl(urlparse(incoming_url).query)
+            
+            # Create list of matching params
+            matching_params = list(set(exp_params) & set(inc_params))
+            
+            # Update list order for expected params
+            exp_params = list(set(exp_params))
+            
+            if matching_params and exp_params:
+                return matching_params==exp_params
+            
+            return False
+        
+        def check_urls_match(self, expected_url, incoming_url):
+            
+            # Check URL paths
+            paths_match = self.check_paths_match(expected_url, incoming_url)
+            
+            # Check URL params
+            params_match = self.check_params_match(expected_url, incoming_url)
+            
+            if paths_match and params_match:
+                return paths_match==params_match
+            
+            return False
```

Comment:
`paths_match` and `params_match` are both `boolean` values.
So if `paths_match and params_match` is `True`, `paths_match == params_match` is also always `True`.
Short form would be `return paths_match and params_match`

Reply:
Good catch. It was also an inaccurate method for performing the multi-boolean comparison since False could've returned True if it weren't for the if block. Fixed and pushed.